tools/topology/topology2/doc/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.13)
+
+project(SOF_TOPOLOGY2_DOC NONE)
+
+set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
+
+set(top_srcdir "${SOF_ROOT_SOURCE_DIRECTORY}")
+
+configure_file(
+	"${PROJECT_SOURCE_DIR}/sof.doxygen.in"
+	"${PROJECT_BINARY_DIR}/sof.doxygen"
+)
+
+add_custom_command(
+	OUTPUT ${PROJECT_SOURCE_DIR}/extra-contents/contents.doxy
+	COMMAND bash ${PROJECT_SOURCE_DIR}/topology2-generate-contents.sh
+	VERBATIM
+)
+
+add_custom_target("doc" ALL
+	COMMAND doxygen sof.doxygen
```

Comment:
Don't use COMMAND in `add_custom_target()`. For EACH custom COMMAND you need:
- one `add_custom_command()` where the COMMAND is
- one `add_custom_target()` without any COMMAND.